circuit_database/plonky2/merkle_tree/src/lib.rs

```diff
@@ -0,0 +1,182 @@
+use plonky2::plonk::proof::ProofWithPublicInputs;
+use plonky2::{
+    field::{goldilocks_field::GoldilocksField, types::Field},
+    hash::{
+        hash_types::{HashOut, HashOutTarget, RichField},
+        poseidon::PoseidonHash,
+    },
+    iop::witness::{PartialWitness, WitnessWrite},
+    plonk::{
+        circuit_builder::CircuitBuilder,
+        circuit_data::{CircuitConfig, CircuitData, CommonCircuitData, VerifierOnlyCircuitData},
+        config::{GenericConfig, PoseidonGoldilocksConfig},
+        proof,
+    },
+};
+
+mod merkle_tree;
+use merkle_tree::MerkleTree;
+
+use serde::{Deserialize, Serialize};
+use std::fs;
+
+pub const D: usize = 2;
+pub type C = PoseidonGoldilocksConfig;
+pub type F = <C as GenericConfig<D>>::F;
+
+#[derive(Serialize, Deserialize)]
+pub struct InputData {
+    pub inputs: Vec<u64>,
+    pub index: usize,
+}
+
+impl InputData {
+    pub fn from_json(path: &str) -> Self {
+        let contents = fs::read_to_string(path).expect("Something went wrong reading the file");
+        serde_json::from_str(&contents).unwrap()
+    }
+    pub fn set_leaf_index(&mut self, index: usize) {
+        // make sure that the user doesn't select an index that is out of bounds
+        assert!(index < self.inputs.len());
+        self.index = index;
+    }
+
+    pub fn get_layers(&self) -> usize {
+        let total_leaves = self.inputs.len();
+        // while total_leaves > 1 {
+        //     total_leaves = (total_leaves + 1) / 2;
+        //     layers += 1;
+        // }
+        let layers = ((total_leaves as f32).log2()) as usize;
+        layers
+    }
+}
+
+pub struct MerkleTreeCircuit {
+    pub proof: ProofWithPublicInputs<F, C, D>,
+    pub verifier_only: VerifierOnlyCircuitData<C, D>,
+    pub common: CommonCircuitData<F, D>,
+}
+
+impl MerkleTreeCircuit {
+    pub fn prove(path: &str) -> Self {
+        //let mut targets: Vec<plonky2::hash::hash_types::HashOutTarget> = Vec::new();
```

Comment:
commented code?

Reply:
Removed in commit: af5823accd0fb1333553d8a8fad0cbfa336b13d8

Reply:
No, this commented code is not removed. Please check the `circuit_database` path.